test/e2e/framework/kubectl_util.go

```diff
@@ -0,0 +1,380 @@
+/*
```

Comment:
this should be it's own sub-package under framework otherwise everyone will inherit and I doubt most folks vendoring the framework would care to take this dependency. 

/hold

Reply:
Good point, I'll update that

Reply:
Still working on this. Having issues with circular dependencies, when taking `KubectlCmd` out of `utils`.

Have to figure out a way to deal with needing `TestContext` and `ProviderInterface` in the `kubectl_util` file. Keeps leading back to depending on the `framework` package.

Reply:
I think I may have figured out a way to do it. Should be able to make a PR shortly.